verticapy/tests_new/performance/vertica/test_conn_simple.py

```diff
@@ -0,0 +1,105 @@
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless  required  by applicable  law or  agreed to in
+writing, software  distributed  under the  License is
+distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+OR CONDITIONS OF ANY KIND, either express or implied.
+See the  License for the specific  language governing
+permissions and limitations under the License.
+"""
+import pytest
+import logging
+
+import verticapy as vp
+import vertica_python
+
+from verticapy.performance.vertica import QueryProfiler
+
+from io import StringIO
+
+
+def test_profile_simple(vp_connect_get_name, raw_client_cursor):
+    """Create a query profiler and run the steps on a simple query"""
+
+    table_name = "test_profile_simple"
+    setup_dummy_table(raw_client_cursor, table_name)
+    request = f"SELECT count(*) from {table_name};"
+    qp = QueryProfiler(request)
+    check_version(qp)
+    check_request(qp, "SELECT COUNT(*)")
+
+
+def setup_dummy_table(raw_client_cursor, table_name):
+    raw_client_cursor.execute(f"DROP table if exists {table_name};")
+    raw_client_cursor.fetchall()
+    raw_client_cursor.execute(f"CREATE TABLE {table_name} (x int);")
+    raw_client_cursor.fetchall()
+
+    raw_client_cursor.executemany(
+        f"insert into {table_name} values (?);",
+        [(1,), (2,), (3,), (4,)],
+        use_prepared_statements=True,
+    )
+    raw_client_cursor.fetchall()
+
+
+def check_version(qp):
+    version_tuple = qp.get_version()
+    logging.info(f"Version is: {version_tuple}")
+    # version tuple can be
+    #  (24, 1, 0)    (dev build)
+    #  (11, 0, 1, 2) (release build)
+    assert len(version_tuple) == 3 or len(version_tuple) == 4
+    assert version_tuple[0] >= 23 or version_tuple[0] in [12, 11]
+
+
+def check_request(qp, fragment):
+    sql = qp.get_request(indent_sql=False)
+    # sql won't match the input query exactly
+    # input query = select count(*) from foo;
+    # stored query = PROFILE select count(*) from foo;
+    logging.info(f"Request retreived is: {sql}")
+    assert fragment.lower() in sql.lower()
+
+
+def test_connect_perf(vp_connect_get_name, raw_client_cursor):
```

Comment:
@jslaunwhite-microfocus - it would be helpful to add pytest.mark.parametrize as there are multiple assert.

Reply:
This has been removed. 

As discussed, parameterized isn't exactly what we're looking for here. Instead, the sequence of asserts confirms that the object behaves as expected.